ts/components/conversation/CreateGroupDialog.tsx

```diff
@@ -0,0 +1,212 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Contact, MemberList } from './MemberList';
+
+declare global {
+  interface Window {
+    Lodash: any;
+    doCreateGroup: any;
+    SMALL_GROUP_SIZE_LIMIT: number;
+  }
+}
+
+interface Props {
+  titleText: string;
+  okText: string;
+  cancelText: string;
+  friendList: Array<any>;
+  i18n: any;
+  onClose: any;
+}
+
+interface State {
+  friendList: Array<Contact>;
+  groupName: string;
+  errorDisplayed: boolean;
+  errorMessage: string;
+}
+
+export class CreateGroupDialog extends React.Component<Props, State> {
+  constructor(props: any) {
+    super(props);
+
+    // const _ = window.Lodash;
+
+    this.onMemberClicked = this.onMemberClicked.bind(this);
+    this.onClickOK = this.onClickOK.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.closeDialog = this.closeDialog.bind(this);
+    this.onGroupNameChanged = this.onGroupNameChanged.bind(this);
+
+    let friends = this.props.friendList;
+    friends = friends.map(d => {
+      const lokiProfile = d.getLokiProfile();
+      const name = lokiProfile ? lokiProfile.displayName : 'Anonymous';
+
+      return {
+        id: d.id,
+        authorPhoneNumber: d.id,
+        authorProfileName: name,
+        selected: false,
+        authorName: name, // different from ProfileName?
+        authorColor: d.getColor(),
+        checkmarked: false,
+      };
+    });
+
+    this.state = {
+      friendList: friends,
+      groupName: '',
+      errorDisplayed: false,
+      // if empty, the initial height is 0, which is not desirable
+      errorMessage: 'placeholder',
+    };
+
+    window.addEventListener('keyup', this.onKeyUp);
+  }
+
+  public onClickOK() {
+    const members = this.state.friendList
+      .filter(d => d.checkmarked)
+      .map(d => d.id);
+
+    if (!this.state.groupName.trim()) {
+      this.onShowError(this.props.i18n('emptyGroupNameError'));
+
+      return;
+    }
+
+    window.doCreateGroup(this.state.groupName, members);
+
+    this.closeDialog();
+  }
+
+  public render() {
+    const checkMarkedCount = this.getMemberCount();
+
+    const titleText = `${this.props.titleText} (Members: ${checkMarkedCount})`;
+    const okText = this.props.okText;
+    const cancelText = this.props.cancelText;
+
+    const errorMessageClasses = classNames(
+      'error-message',
+      this.state.errorDisplayed ? 'error-shown' : 'error-faded'
+    );
+
+    return (
+      <div className="content">
+        <p className="titleText">{titleText}</p>
+        <p className={errorMessageClasses}>{this.state.errorMessage}</p>
+        <input
+          type="text"
+          id="group-name"
+          className="group-name"
+          placeholder="Group Name"
```

Comment:
Nit pick but everywhere we use raw English strings like this instead of the i18n thing is going to come back to bite us when we want to support translations

Reply:
updated